check/utils.go

```diff
@@ -258,6 +258,36 @@ func makeFilename(fn string) string {
 	return fn
 }
 
+func generateFileSummaryMap(out *bufio.Scanner, dir string, fsMap map[string]FileSummary) error {
```

Comment:
@tariq1890 rather than updating the passed-in `fsMap`, could we have this function return `map[string]FileSummary, error` instead?